--- a/net/minecraft/client/gui/GuiTextField.java
+++ b/net/minecraft/client/gui/GuiTextField.java
@@ -2,8 +2,11 @@
 
 import net.minecraft.client.renderer.Tessellator;
 import net.minecraft.util.ChatAllowedCharacters;
+
 import org.lwjgl.opengl.GL11;
 
+import com.mtbs3d.minecrift.provider.MCOpenVR;
+
 public class GuiTextField extends Gui
 {
     private final FontRenderer fontRendererInstance;
@@ -673,6 +676,11 @@
         }
 
         this.isFocused = p_146195_1_;
+        
+        // VIVE START - show keyboard
+        System.out.println("Text element focused " + p_146195_1_);
+        MCOpenVR.setKeyboardOverlayShowing(p_146195_1_);
+        // VIVE END - show keyboard
     }
 
     /**
